src/pkg/packager/sources/tarball.go

```diff
@@ -0,0 +1,300 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package sources contains core implementations of the PackageSource interface.
+package sources
+
+import (
+	"archive/tar"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/defenseunicorns/zarf/src/pkg/layout"
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	"github.com/defenseunicorns/zarf/src/pkg/oci"
+	"github.com/defenseunicorns/zarf/src/pkg/utils"
+	"github.com/defenseunicorns/zarf/src/types"
+	"github.com/mholt/archiver/v3"
+)
+
+var (
+	// veryify that TarballSource implements PackageSource
+	_ PackageSource = (*TarballSource)(nil)
+
+	// veryify that SplitTarballSource implements PackageSource
+	_ PackageSource = (*SplitTarballSource)(nil)
```

Comment:
It would be nice to keep sources to 1 per file - the split source can still call tarball source functions.